edelphi/src/main/java/fi/metatavu/edelphi/binaries/queries/QueryCommentExportBinaryController.java

```diff
@@ -0,0 +1,107 @@
+import java.util.logging.Logger;
+
+import org.apache.commons.codec.binary.Base64;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import fi.metatavu.edelphi.EdelfoiStatusCode;
+import fi.metatavu.edelphi.binaries.BinaryController;
+import fi.metatavu.edelphi.dao.panels.PanelStampDAO;
+import fi.metatavu.edelphi.dao.querydata.QueryReplyDAO;
+import fi.metatavu.edelphi.dao.resources.QueryDAO;
+import fi.metatavu.edelphi.domainmodel.panels.PanelStamp;
+import fi.metatavu.edelphi.domainmodel.querydata.QueryReply;
+import fi.metatavu.edelphi.domainmodel.resources.Query;
+import fi.metatavu.edelphi.i18n.Messages;
+import fi.metatavu.edelphi.pages.panel.admin.report.util.QueryReplyFilter;
+import fi.metatavu.edelphi.pages.panel.admin.report.util.ReportContext;
+import fi.metatavu.edelphi.smvcj.SmvcRuntimeException;
+import fi.metatavu.edelphi.smvcj.controllers.BinaryRequestContext;
+import fi.metatavu.edelphi.utils.QueryDataUtils;
+import fi.metatavu.edelphi.utils.QueryDataUtils.ReplierExportStrategy;
+import fi.metatavu.edelphi.utils.ResourceUtils;
+
+public class QueryCommentExportBinaryController extends BinaryController {
+
+  private static Logger logger = Logger.getLogger(QueryCommentExportBinaryController.class.getName());
+
+  @Override
+  public void process(BinaryRequestContext requestContext) {
+    QueryDAO queryDAO = new QueryDAO();
+    QueryReplyDAO queryReplyDAO = new QueryReplyDAO();
+    PanelStampDAO panelStampDAO = new PanelStampDAO();
+    
+    Long queryId = requestContext.getLong("queryId");
+    Long stampId = requestContext.getLong("stampId");
+
+    ExportFormat format = ExportFormat.valueOf(requestContext.getString("format"));
+    if (format == null) {
+      throw new SmvcRuntimeException(EdelfoiStatusCode.QUERYDATAEXPORTFAILURE, "Missing parmeter format");
+    }
+
+    ReportContext reportContext = null;
+    String serializedContext = requestContext.getString("serializedContext");
+    if (serializedContext != null) {
+      try {
+        ObjectMapper objectMapper = new ObjectMapper();
+        byte[] serializedData = Base64.decodeBase64(serializedContext);
+        String stringifiedData = new String(serializedData, "UTF-8");
+        reportContext = objectMapper.readValue(stringifiedData, ReportContext.class); 
+      } catch (Exception e) {
+        logger.log(Level.SEVERE, "Failed to create serialized context", e);
+      }
+    }
+    
+    Query query = queryDAO.findById(queryId);
+    PanelStamp panelStamp = panelStampDAO.findById(stampId);
+
+    // Query replies, possibly filtered
+        
+    List<QueryReply> replies = queryReplyDAO.listByQueryAndStampAndArchived(query, panelStamp, Boolean.FALSE);
+    List<QueryReplyFilter> filters = reportContext == null ? null : QueryReplyFilter.parseFilters(reportContext.getFilters());
+    if (filters != null) {
+      for (QueryReplyFilter filter : filters) {
+        replies = filter.filterList(replies);
+      }
+    }
+
+    switch (format) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)